validator/client/grpc-api/grpc_validator_client.go

```diff
@@ -23,8 +24,60 @@ type grpcValidatorClient struct {
 	isEventStreamRunning      bool
 }
 
-func (c *grpcValidatorClient) Duties(ctx context.Context, in *ethpb.DutiesRequest) (*ethpb.DutiesResponse, error) {
-	return c.beaconNodeValidatorClient.GetDuties(ctx, in)
+func (c *grpcValidatorClient) Duties(ctx context.Context, in *ethpb.DutiesRequest) (*ethpb.ValidatorDutiesContainer, error) {
+	dutiesResponse, err := c.beaconNodeValidatorClient.GetDuties(ctx, in)
+	if err != nil {
+		return nil, err
+	}
+	return toValidatorDutiesContainer(dutiesResponse)
+}
+
+func toValidatorDutiesContainer(dutiesResponse *ethpb.DutiesResponse) (*ethpb.ValidatorDutiesContainer, error) {
+	currentDuties := make([]*ethpb.ValidatorDuty, len(dutiesResponse.CurrentEpochDuties))
+	for i, cd := range dutiesResponse.CurrentEpochDuties {
+		duty, err := toValidatorDuty(cd)
+		if err != nil {
+			return nil, err
+		}
+		currentDuties[i] = duty
+	}
+	nextDuties := make([]*ethpb.ValidatorDuty, len(dutiesResponse.NextEpochDuties))
+	for i, nd := range dutiesResponse.NextEpochDuties {
+		duty, err := toValidatorDuty(nd)
+		if err != nil {
+			return nil, err
+		}
+		nextDuties[i] = duty
+	}
+	return &ethpb.ValidatorDutiesContainer{
+		CurrentEpochDuties: currentDuties,
+		NextEpochDuties:    nextDuties,
+	}, nil
+}
+
+func toValidatorDuty(duties *ethpb.DutiesResponse_Duty) (*ethpb.ValidatorDuty, error) {
+	var valIndexInCommittee uint64
+	// valIndexInCommittee will be 0 in case we don't get a match. This is a potential false positive,
+	// however it's an impossible condition because every validator must be assigned to a committee.
+	for cIndex, vIndex := range duties.Committee {
+		if vIndex == duties.ValidatorIndex {
+			valIndexInCommittee = uint64(cIndex)
+			break
+		}
+	}
+	return &ethpb.ValidatorDuty{
+		CommitteeLength:         uint64(len(duties.Committee)),
+		CommitteeIndex:          duties.CommitteeIndex,
+		CommittesAtSlot:         duties.CommitteesAtSlot, // GRPC doesn't use this value though
+		ValidatorCommitteeIndex: valIndexInCommittee,
+		AttesterSlot:            duties.AttesterSlot,
+		ProposerSlots:           duties.ProposerSlots,
```

Comment:
do i need to safecopy here because it's a slice?

Reply:
No, because it's very unlikely someone will modify this value, as it's just a "read-only" response from the node. I see you copied the public key, so maybe let's be consistent and either copy both or neither.